### ISSUE TITLE
Plugin throws exception sometimes (IntelliJ IDEA 2023.3.2; ANSI Highlighter Premium 23.3.1)

### ISSUE THREAD

**Issue by ofry** [niftythoughts/ansi-highlighter-premium#14]
ANSI Highlighter Premium 23.3.1

```
IntelliJ IDEA 2023.3.2 (Ultimate Edition)
Build #IU-233.13135.103, built on December 20, 2023
Runtime version: 17.0.9+7-b1087.9 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 6.5.0-14-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 10240M
Cores: 4
Registry:
  ide.images.show.chessboard=true
Non-Bundled Plugins:
  com.jetbrains.plugins.ini4idea (233.13135.116)
  Batch Scripts Support (1.0.13)
  icu.windea.bbcode (1.8)
  dev.meanmail.plugin.nginx-intellij-plugin-pro (2022.1.1)
  intellij-awk (0.4.3)
  com.plugin.image-icon-viewer (1.0.11)
  com.github.obiscr.exceleditor (2023.11.1-222.233)
  nl.bryanderidder.regexrenamefiles (1.2.7)
  com.alayouni.ansiHighlight (23.3.1)
  org.intellij.RegexpTester (2.2.0)
  com.github.holgerbrandl.pasteimages/ (1.3.2)
  ru.limydesign.plugins.yandex.translate (0.1.0)
  artsiomch.cmake (233.1.0)
  artsiomch.cmake.plus (233.1.0)
  com.mnw.tabmover (1.5.2)
  ru.ilimurzin.bitrixexcluder (1.1.0)
  com.bruce.intellijplugin.RedisPlugin (2023.10.28)
  com.github.izhangzhihao.intellijgooglesearch (0.0.2)
  com.github.exidcuter.dockerregistryexplorer (1.1.0-stable)
  com.jetbrains.CyanTheme (1.3)
  com.intellij.apacheConfig (233.13135.116)
  com.vexus2.intellij.openingithub (1.0.6)
  org.jetbrains.plugins.stylus (233.11799.172)
  com.mroche.JsonHelper (1.3.2)
  de.ehmkah.projects.imgdiff (1.11.0)
  org.jetbrains.plugins.localization (233.11799.188)
  com.devwook.copy_path_helper (1.1.2)
  lermitage.intellij.extra.icons (2024.2.2)
  krasa.IdeaIconPack (1.14)
  man-page-viewer (0.1.1)
  MHCommitMessageTemplate (1.0.3)
  String Manipulation (9.12.0)
  com.jetbrains.packagesearch.intellij-plugin (233.11799.196)
  com.euphoricity.gitignore (0.3)
  GrepConsole (12.22.1-IJ2021.1)
  com.intellij.ml.llm (233.13135.116)
  com.majera.intellij.codereview.bitbucket (2024.1.1.233)
  git-extended-update-index (0.1.3)
  mobi.hsz.idea.gitignore (4.5.2)
  com.intellij.tide (233.13135.116)
  com.ess.terminal-pro (1.1.7)
  Pythonid (233.13135.103)
  net.fhtagn.pycharm.cellmode (1.3.1)
  io.github.guoci.PythonDocumentationLinkProvider (0.1.3)
  org.tonybaloney.security.pycharm-security (1.29.0)
  ru.meanmail.plugins.prettify-python (2022.1)
  ru.meanmail.plugin.django_command_runner (2022.1)
  com.github.kawamataryo.copygithublink (0.5.3)
  com.majera.intellij.codereview.gitlab (2024.1.1.233)
  zielu.gittoolbox (500.1.8+222)
  com.jetbrains.php (233.13135.108)
  com.niclas-van-eyk.laravel-make-integration (2.3.5)
  com.jetbrains.twig (233.13135.116)
  com.xu (1.3)
  com.funivan.idea.phpClean (2023.12.17)
  com.clutcher.comments_highlighter (2024.1.2)
  com.bitrixsoft.lang (1.2.2)
  nl.deschepers.laraveltinker (2.6.1)
  ru.itrack.bitrix (0.2.0)
  com.kalessil.phpStorm.phpInspectionsUltimate-mp (2024.1)
  com.jetbrains.php.blade (233.13135.108)
  dev.ekvedaras.laravelquery (4.0.2)
  com.kalessil.phpStorm.yii2inspections (1.0.6)
  org.jetbrains.plugins.phpstorm-remote-interpreter (233.11799.172)
  org.jetbrains.plugins.phpstorm-docker (233.13135.67)
  com.jetbrains.php.framework (233.13135.116)
  com.jetbrains.php.wordPress (233.13135.108)
  ru.vkurdin.idea.php.lambdafolding (0.3.0)
  org.exbin.deltahex.intellij (0.2.9)
  pers.fw.doc_en (1.1)
  com.wbars.php.folding (1.0.3)
  com.intellij.php.codeigniter3 (0.6.0)
  com.github.woru.options-completion-phpstorm-plugin (0.0.7)
  ru.adelf.idea.dotenv (2023.3)
  com.laravel_idea.plugin (8.0.4.233)
  org.sourcelab.intellij.plugin.PHPDataObjectGenerator (2021.1.16)
  com.eny.i18n (2.6.2)
  net.king2500.plugins.PhpAdvancedAutoComplete (1.1.0)
  de.espend.idea.php.toolbox (6.2.0)
  de.espend.idea.php.annotation (9.4.0)
  fr.adrienbrault.idea.symfony2plugin (2022.1.263)
  de.espend.idea.shopware (4.4.3)
  org.endpointdeps.EndpointDependencies (1.3)
  com.firsttimeinforever.intellij.pdf.viewer.intellij-pdf-viewer (0.15.0)
  Base64 for IDEA and Storm (1.1b)
Kotlin: 233.13135.103-IJ
Current Desktop: KDE
```

```
java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.base/java.util.Objects.checkIndex(Objects.java:361)
	at java.base/java.util.ArrayList.get(ArrayList.java:427)
	at cm.nt.ah.bj.a(SourceFile:142)
	at cm.nt.ah.bj.b(SourceFile:69)
	at cm.nt.ah.bl.mouseDragged(SourceFile:38)
	at java.desktop/java.awt.AWTEventMulticaster.mouseDragged(AWTEventMulticaster.java:328)
	at java.desktop/java.awt.Component.processMouseMotionEvent(Component.java:6705)
	at java.desktop/javax.swing.JComponent.processMouseMotionEvent(JComponent.java:3403)
	at java.desktop/java.awt.Component.processEvent(Component.java:6426)
	at java.desktop/java.awt.Container.processEvent(Container.java:2266)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5027)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4855)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4969)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4600)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4524)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2809)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4855)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:794)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:766)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:764)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:763)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:695)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.kt:635)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$10(IdeEventQueue.kt:580)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithEnabledImplicitRead(RwLockHolder.kt:75)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithImplicitRead(RwLockHolder.kt:67)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:580)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:72)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:355)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:354)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:793)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:354)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:1014)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:114)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1014)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:349)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:848)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:391)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
```

**Comment by niftythoughts** [comment]
Thanks for reporting, can you provide steps to reproduce the issue?

**Comment by ofry** [comment]
No, sorry... I don't remember exast steps to reproduce it, because it happens rarely.

**Comment by niftythoughts** [comment]
> No, sorry... I don't remember exast steps to reproduce it, because it happens rarely.

Difficult to identify the cause without enough context.

Does the plugin stop working when the issue occurs? 

Do you notice any side effects other than the exception?

**Comment by ofry** [comment]
Hm... I will give you some clues.
1) This was happened when file was rewritten externally when I surf this file.
2) Yes, it stopped working. After close and reopen file it worked fine again.
3) No such side-effects.

**Comment by niftythoughts** [comment]
Bug-fix submitted Apr 2nd, should be approved within 48 hours.

Apologies for the inconvenience.

**Comment by niftythoughts** [comment]
Fixed in [24.1.0-RC](https://plugins.jetbrains.com/plugin/9707-ansi-highlighter-premium/versions/stable/515040)
